lib/service/connect.go

```diff
@@ -874,10 +874,17 @@ func tunnelAddr(settings client.ProxySettings) (string, error) {
 
 func (process *TeleportProcess) newClientThroughTunnel(servers []utils.NetAddr, identity *auth.Identity) (*auth.Client, error) {
 	// Discover address of SSH reverse tunnel server.
-	proxyAddr, err := process.findReverseTunnel(servers)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	proxyAddr := ""
+	if process.Config.SSH.ReuseConnIotMode == true {
+		proxyAddr = servers[0].String()
+	} else {
+		ret, err := process.findReverseTunnel(servers)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		proxyAddr = ret
```

Comment:
Instead of adding an additional flag, can you change the logic to the following:

1. Try to discover the address of the reverse tunnel (call `process.findReverseTunnel`).
2. If that fails, add a log line explaining that the client was unable to discover the address of reverse tunnel and will attempt to fallback to the address in file configuration (include the address in the log line). In a comment in source, please explain this will only work if you are multiplexing web and reverse tunnel on the same port (`3080` and `3024`).